jax/_src/api.py

```diff
@@ -3159,6 +3159,57 @@ def named_call_f(*args, **kwargs):
 
   return named_call_f
 
+@contextmanager
+def named_scope(
+    name: str,
+  ) -> Generator[None, None, None]:
+  """A context manager that adds a user specified name to the JAX name stack.
+
+  When staging out computations for just-in-time compilation to XLA (or other
+  backends such as TensorFlow) JAX runs your Python program but by default does
+  not preserve any of the function names or other metadata associated with it.
```

Comment:
```suggestion
  backends such as TensorFlow) JAX does not, by default, preserve the names
  (or other source metadata) of Python functions it encounters.
```
Trying to avoid an unnecessary "JAX runs your program", to skip any incidental confusion regarding (abstract vs. concrete) evaluation.